pkg/reconciler/v1alpha1/taskrun/entrypoint/entrypoint.go

```diff
@@ -43,41 +43,36 @@ const (
 	MarkerFile        = "/tools/marker-file.txt"
 )
 
+var entrypointCacheSize = 1024
```

Comment:
let's make this a const value instead - I think you have made this a variable so that the unit tests can change it, but I recommend that we change this to one of:

1. Use [the cache interface directly](https://godoc.org/github.com/hashicorp/golang-lru#New), without wrapping it, but have this a public const that users can call, e.g. the reconciler can call `lru.New(entrypoint.CacheSize)`
2. Keep the `NewCache` wrapper function that you have created, use a private const `cacheSize` inside that function, and call `lru.New(entrypoint.cacheSize` in the tests (it's okay to skip test coverage for `NewCache` if we can avoid mutable global state)
3. Keep everything as is, but use the same cache size (1024) in the tests themselves (unless this is really slow, which would surprise me), e.g.:
```
	for i := 0; i < entrypoint.cacheSize + 1; i++ {
		image := fmt.Sprintf("image%d:latest", i)
		entrypointCache.set(image, entrypoint)
	}
	for i := 0; i < entrypoint.cacheSize; i++ {
		image := fmt.Sprintf("image%d:latest", i)
		if _, ok := entrypointCache.get(image); ok {
			t.Errorf("entrypoint of image %s should be expired", image)
		}
	}
	for i := entrypoint.cacheSize; i < entrypoint.cacheSize + 1; i++ {
		image := fmt.Sprintf("image%d:latest", i)
		if _, ok := entrypointCache.get(image); !ok {
			t.Errorf("entrypoint of image %s shouldn't be expired", image)
		}
	}
```
4. (Or anything else you can think of that avoids mutable global state)

This makes it easier to understand both the code and the tests, and avoids having mutable global state.

Reply:
The UT is fast. Just choose the third one.